### ISSUE TITLE
AxiosInstance cannot be invoked in TypeScript

### ISSUE THREAD

**Issue by evanshortiss** [axios/axios#1253]
## Summary

Currently the following will fail in TypeScript. 

```ts
const axios = require('axios')
const instance = axios.create(DEFAULT_CONFIG)

instance(REQUEST_CONFIG)
  .then(onSuccess)
  .catch(onError)
```

## Context

- axios version: 0.17.1
- Environment: node.js v6.9.1, typescript 2.6.2

**Comment by evanshortiss** [comment]
Addressed by #1254

**Comment by nickuraltsev** [comment]
Fixed via #1192 and #1254
